Fixed the type definition of SceneUpdateCallback

arosemena opened the pull request:
Please do not update the README or Change Log, we will do this when we merge your PR.

This PR (delete as applicable)

* Fixes a bug

Describe the changes below:

Fixes the `SceneUpdateCallback` type definition, it was missing the `time` and `delta` parameters

arosemena committed:
Fixed the type definition of SceneUpdateCallback

photonstorm commented:
Thank you for spending time contributing towards the Phaser TypeScript Defs, I really appreciate it. However, I'm afraid I cannot merge your PR. The reason is that you've edited the Phaser defs file directly. Our defs file is generated automatically as part of our build process, it works by running a custom parser across the Phaser source code, extracting the JSDoc blocks and then building the types from those. This means, if we were to merge your PR, your changes would be instantly lost the next time the defs are generated.

In order to fix the defs permanently, you need to edit the underlying JSDocs in the source code that is causing the defs error. Sometimes, the error is a result of a bug in the TS parser, but mostly it's because of incorrectly typed JSDocs. Please look at the command `npm run help` to see the options available to you re: building and running our TS parser for yourself, so you can edit the JSDoc blocks, run the parser and see if the end result is resolved or not.

arosemena committed:
Added time parameters to SceneUpdateCallback typedef file

arosemena committed:
Reverted default phaser.d.ts file

arosemena commented:
Apologies @photonstorm I missed that part, I've gone ahead and updated the correct typedef file and reverted the changes done directly at `phaser.d.ts`, let me know if there's anything else I should add, thanks!

photonstorm commented:
@arosemena thanks - I wonder if we should make both parameters optional? As this change will impact lots of TS devs otherwise.

arosemena commented:
Types should still be compatible, extra parameters can be ignored because we're using this from the consumer side, if phaser is always passing those values then they shouldn't be optional as we know they'll be there and having them as optional would require additional work to check they are defined which would be redundant because we know they are there.

Here's a sample where you can confirm this:

```typescript
type Before = (this: Scene) => void
type After = (this: Scene, time: number, delta: number) => void

const oldUpdate: Before = function() {} // valid old code
const newUpdate: After = oldUpdate // no type error
const newUpdate2: After = function(time, delta) {} // no type error, this would not work with the old type
```

photonstorm commented:
👍
